### PR TITLE
Adding new API endpoint prices

### PR DESCRIPTION
**Pull request by Martin-Molinero**
#### Description

- Adding new API endpoint prices
    - Adding unit tests
- Adding new `DoubleUnixSecondsDateTimeJsonConverter` which defines a `JsonConverter` that serializes `DateTime`, uses the number of whole and fractional seconds since unix epoch
#### Related Issue

Closes https://github.com/QuantConnect/Lean/issues/2333
#### Motivation and Context

It's difficult to simultaneously add a new asset to the portfolio and get the current price of that asset. Few brokerages have an API which supports this easily.
#### Requires Documentation Change

N/A
#### How Has This Been Tested?

Unit test
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Common/API/Prices.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+using Newtonsoft.Json;
+using QuantConnect.Api;
+using QuantConnect.Util;
+
+namespace QuantConnect.API
+{
+    /// <summary>
+    /// Prices rest response wrapper
+    /// </summary>
+    public class Prices : RestResponse
```
Comment (mchandschuh) [inline Common/API/Prices.cs]: will we get `success` in each `prices` element?

File: Tests/API/ApiTests.cs [hunk 2]
```diff
@@ -596,6 +596,47 @@ public void BacktestingData_CanBeDownloadedAndSaved_Successfully()
...
+
+            Assert.IsTrue(pricesList.Success);
+            Assert.AreEqual(pricesList.Prices.Count, 1);
+
+            var price = pricesList.Prices.First();
+            Assert.AreEqual(price.Symbol, spy.ID.ToString());
+            Assert.AreNotEqual(price.Price, 0);
+            var updated = price.Updated;
+            var reference = DateTime.UtcNow.Subtract(TimeSpan.FromDays(3));
+            Assert.IsTrue(updated > reference);
```
Comment (mchandschuh) [inline Tests/API/ApiTests.cs]: interesting assert condition, these are integrations tests so it's all gravy

File: Tests/API/ApiTests.cs [hunk 3]
```diff
@@ -596,6 +596,47 @@ public void BacktestingData_CanBeDownloadedAndSaved_Successfully()
...
+
+            Assert.IsTrue(pricesList.Success);
+            Assert.AreEqual(pricesList.Prices.Count, 2);
+
+            Assert.IsFalse(pricesList.Prices.Any(x => x.Price == 0));
+            Assert.AreEqual(pricesList.Prices.Count(x => x.Symbol == aapl.ID.ToString()), 1);
+            Assert.AreEqual(pricesList.Prices.Count(x => x.Symbol == spy.ID.ToString()), 1);
+
+            var reference = DateTime.UtcNow.Subtract(TimeSpan.FromDays(3));
+            Assert.IsFalse(pricesList.Prices.Any(x => x.Updated < reference));
```
Comment (mchandschuh) [inline Tests/API/ApiTests.cs]: double negative? perhaps `Assert.IsTrue(priceList.Prices.All(x => x.Updated > reference)` is more intutive?

File: Common/API/Prices.cs (hunk 1 above)
Comment (Martin-Molinero) [reply Common/API/Prices.cs]: no 😖. It actually doesn't make sense for `Prices` to inherit from `RestResponse` anymore, I'll remove it.

File: Tests/API/ApiTests.cs (hunk 3 above)
Comment (Martin-Molinero) [reply Tests/API/ApiTests.cs]: Yeah! I'll change it 👍

File: Tests/API/ApiTests.cs (hunk 2 above)
Comment (Martin-Molinero) [reply Tests/API/ApiTests.cs]: Yeah, my thought was it should pass even if you run it on a Sunday

**Pull request merged**